Tasks/AzureCLIV3/azureclitask.ts

```diff
@@ -0,0 +1,469 @@
+
+        if (authScheme.toLowerCase() == "workloadidentityfederation") {
+            await this.loginWithWorkloadIdentityFederation(connectedService, visibleAzLogin);
+        }
+        else if (authScheme.toLowerCase() == "serviceprincipal") {
+            let authType: string = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
+            let cliPassword: string = null;
+            let authParam: string = "--password";
+            var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
+            var tenantId: string = tl.getEndpointAuthorizationParameter(connectedService, "tenantid", false);
+
+            this.servicePrincipalId = servicePrincipalId;
+            this.tenantId = tenantId;
+
+            if (authType == "spnCertificate") {
+                tl.debug('certificate based endpoint');
+                if(this.isSupportCertificateParameter) {
+                    authParam = "--certificate";
+                }
+                let certificateContent: string = tl.getEndpointAuthorizationParameter(connectedService, "servicePrincipalCertificate", false);
+                cliPassword = path.join(tl.getVariable('Agent.TempDirectory') || tl.getVariable('system.DefaultWorkingDirectory'), 'spnCert.pem');
+                fs.writeFileSync(cliPassword, certificateContent);
+                this.cliPasswordPath = cliPassword;
+            }
+            else {
+                tl.debug('key based endpoint');
+                cliPassword = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+                this.servicePrincipalKey = cliPassword;
+            }
+
+            let escapedCliPassword = cliPassword.replace(/"/g, '\\"');
+            tl.setSecret(escapedCliPassword.replace(/\\/g, '\"'));
+            //login using svn
+            if (visibleAzLogin) {
+                Utility.throwIfError(tl.execSync("az", `login --service-principal -u "${servicePrincipalId}" ${authParam}="${escapedCliPassword}" --tenant "${tenantId}" --allow-no-subscriptions`), tl.loc("LoginFailed"));
+            }
+            else {
+                Utility.throwIfError(tl.execSync("az", `login --service-principal -u "${servicePrincipalId}" ${authParam}="${escapedCliPassword}" --tenant "${tenantId}" --allow-no-subscriptions --output none`), tl.loc("LoginFailed"));
+            }
+        }
+        else if(authScheme.toLowerCase() == "managedserviceidentity") {
+            //login using msi
+            if (visibleAzLogin) {
+                Utility.throwIfError(tl.execSync("az", "login --identity"), tl.loc("MSILoginFailed"));
+            }
+            else {
+                Utility.throwIfError(tl.execSync("az", "login --identity --output none"), tl.loc("MSILoginFailed"));
+            }            
+        }
+        else {
+            throw tl.loc('AuthSchemeNotSupported', authScheme);
```

Comment:
Should we be more specific and guide user how to solve problem? Right now we just say that scheme is not supported, but it might be good to say what is actually supported.